framework/CMakeLists.txt

```diff
@@ -2,6 +2,9 @@ include("${AREG_BASE}/areg/CMakeLists.txt")
 include("${AREG_BASE}/mcrouter/CMakeLists.txt")
 
 
-# Adding  source files
+# Adding  source files to a single executable.
 add_executable(${PROJECT_NAME} ${areg_SRC} ${mcrouter_SRC})
```

Comment:
Later on, we may need to change the executable name to something like "mcrouter"${AREG_BIN_EXT}
And this target needs to be broken into 2 steps as well:
 - ${areg_SRC} should be built into a library first, this can be a static or a shared lib based on the user input
 - mcrouter then links to the library

Reply:
Thanks @biwiki.
 **Noted**.
I need to tweak *cmake/common.cmake* and *cmake/user.cmake* a little bit more, then it'll be ready to create multiple targets to manage its behavior as you described. (In the next PR this issue will be addressed).

Reply:
Thank you @Ali-Nasrolahi for changes and @biwiki for the review

Good point, @biwiki. In general

1. At the moment the sources support only Windows and Linux. It is planned to support other OS, including RTOS. The target should be specified in a user specific CMake file located in `conf/cmake` folder.
2. The areg library should be compiled as shared or static, which is compiled first. The option is set in user config. 
3. We should consider the case that Windows and Linux have different extensions. At the moment the extensions are EXE or OUT. But different projects may have different policies. For example, some may prefer do no have extension at all and we should give this possibility.

There is a one more thing to keep in mind, it is related with the SIML service interface description files (see in `examples`). The idea is that these files are added in the project (in CMake). The system should automatically call code generator, generate source files and create a static library, which will be added in the project. Right now, either in `make`, `Eclipse` project or `MS Visual Studio` project files these are done manually. But we'll need to change. The details of these I'll explain later.